sdk/package/sdk.ts

```diff
@@ -0,0 +1,726 @@
+        const nonce = json.authorization_signature.signature.nonce_point;
+        json.authorization_signature.signature = [scala, nonce];
+    }
+
+    // transfer receiver_posts to match runtime side
+    json.receiver_posts.map((x:any) => {
+        // `message` is [[[..],[..],[..]]], change to [[..], [..], [..]]
+        var arr1 = x.note.incoming_note.ciphertext.ciphertext.message.flatMap(
+            function(item: any,index:any,a: any){
+            return item;
+        });
+        const tag = x.note.incoming_note.ciphertext.ciphertext.tag; 
+        const pk = x.note.incoming_note.ciphertext.ephemeral_public_key; 
+        x.note.incoming_note.tag = tag;
+        x.note.incoming_note.ephemeral_public_key = pk;
+        x.note.incoming_note.ciphertext = arr1;
+        delete x.note.incoming_note.header;
+
+        const light_pk = x.note.light_incoming_note.ciphertext.ephemeral_public_key; 
+        // ciphertext is [u8; 96] on manta-rs, but runtime side is [[u8; 32]; 3]
+        const light_cipher = x.note.light_incoming_note.ciphertext.ciphertext;
+        const light_ciper0 = light_cipher.slice(0, 32);
+        const light_ciper1 = light_cipher.slice(32, 64);
+        const light_ciper2 = light_cipher.slice(64, 96);
+        x.note.light_incoming_note.ephemeral_public_key = light_pk;
+        x.note.light_incoming_note.ciphertext = [light_ciper0, light_ciper1, light_ciper2];
+        delete x.note.light_incoming_note.header;
+
+        // convert asset value to [u8; 16]
+        x.utxo.public_asset.value = new BN(x.utxo.public_asset.value).toArray('le', 16);
+
+        x.full_incoming_note = x.note;
+        delete x.note;
+    });
+
+    // transfer sender_posts to match runtime side
+    json.sender_posts.map((x:any) => {
+        const pk = x.nullifier.outgoing_note.ciphertext.ephemeral_public_key;
+        const cipher = x.nullifier.outgoing_note.ciphertext.ciphertext; 
+        const ciper0 = cipher.slice(0, 32);
+        const ciper1 = cipher.slice(32, 64);
+        const outgoing = {
+            ephemeral_public_key: pk,
+            ciphertext: [ciper0, ciper1]
+        };
+        x.outgoing_note = outgoing;
+        const nullifier = x.nullifier.nullifier.commitment;
+        x.nullifier_commitment = nullifier;
+        delete x.nullifier;
+    });
+
+    console.log("origin sources:" + JSON.stringify(json.sources));
+    return json
+}
+
+/// Convert uint8Array to number
+/// This method assumes the uint8array is sorted in little-endian form
+/// thus the smallest, least significant value is stored first.
+const uint8ArrayToNumber = (uint8array: AssetId): number => {
+    let value = 0;
+    for (let i = uint8array.length-1; i >= 0; i--) {
+        value = (value * 256) + uint8array[i];
+    }
+    return value;
+}
+
+/// @TODO: Proper implementation of this function
+const numberToUint8Array = (assetIdNumber: number): AssetId => {
+    // @TODO: the `number` type has value limitation, should change to `string` type.
```

Comment:
yeah let's do this

Reply:
Asset I'd and value are handled as arrays of u8 ex. [u8;32] and [u8;16] as runtime expects this as well

Reply:
Yeah, we should change `assetIdNumber` type to `string` or `BN`.

Reply:
We should just make everything on sdk js side compatible either with BN or BigInt and just have conversion to u8 arrays like you said

Reply:
done, change asset_id type to BN.